pkg/discovery/worker/aggregation/container_usage_data_aggregator.go

```diff
@@ -65,18 +61,16 @@ func (maxUsageDataAggregator *maxUsageDataAggregator) String() string {
 	return maxUsageDataAggregator.aggregationStrategy
 }
 
-func (maxUsageDataAggregator *maxUsageDataAggregator) Aggregate(commodities []*proto.CommodityDTO) (float64, float64, float64, error) {
-	if len(commodities) == 0 {
-		err := fmt.Errorf("error to aggregate commodities using %s : commodities list is empty", maxUsageDataAggregator)
+func (maxUsageDataAggregator *maxUsageDataAggregator) Aggregate(resourceMetrics *repository.ContainerMetrics) (float64, float64, float64, error) {
+	if len(resourceMetrics.Used) == 0 {
+		err := fmt.Errorf("error to aggregate container usage data using %s: used data points list is empty", maxUsageDataAggregator)
 		return 0.0, 0.0, 0.0, err
 	}
-	maxCapacity := 0.0
 	maxUsed := 0.0
-	maxPeak := 0.0
-	for _, commodity := range commodities {
-		maxCapacity = math.Max(maxCapacity, *commodity.Capacity)
-		maxUsed = math.Max(maxUsed, *commodity.Used)
-		maxPeak = math.Max(maxPeak, *commodity.Peak)
+	peak := 0.0
+	for _, usedPoint := range resourceMetrics.Used {
+		maxUsed = math.Max(maxUsed, usedPoint.Value)
+		peak = math.Max(peak, usedPoint.Value)
```

Comment:
nit: it looks like maxUsed and peak are guaranteed to always be exactly the same. Is there a good reason to calculate the same value twice?